### PR TITLE
FEATURE: experiment with hot sort order

### PR DESCRIPTION
**Pull request by SamSaffron**

### CODE REVIEW

File: lib/topic_query.rb [hunk 1]
```diff
@@ -338,6 +338,14 @@ def list_bookmarks
     create_list(:bookmarks) { |l| l.where("tu.bookmarked") }
   end
 
+  def list_hot
+    create_list(:hot, unordered: true) do |topics|
```
Comment (pmusaraj) [inline lib/topic_query.rb:342]: I think this list should have a maximum number of results. Not sure exactly what, but 200-300 would be enough IMO.

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -4059,6 +4059,10 @@ en:
         title: "Unseen"
         lower_title: "unseen"
         help: "new topics and topics you are currently watching or tracking with unread posts"
+      hot:
+        title: "Hot"
+        lower_title: "hot"
+        help: "top recent topics"
```
Comment (pmusaraj) [inline config/locales/client.en.yml:4066]:
```suggestion
        help: "hot topics"
```

File: app/models/topic_hot_score.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class TopicHotScore < ActiveRecord::Base
+  belongs_to :topic
+
+  DEFAULT_BATCH_SIZE = 1000
+
+  def self.update_scores(max = DEFAULT_BATCH_SIZE)
+    # score is
+    # (total likes - 1) / (age in hours + 2) ^ gravity
```
Comment (pmusaraj) [inline app/models/topic_hot_score.rb:10]: Should the number of recent replies also influence the score?

**Review by pmusaraj — APPROVED** [review]
Pulled it locally, looks good! 

Added a small commit for the failing spec (RSS route).

File: app/models/topic_hot_score.rb (hunk 3 above)
Comment (SamSaffron) [reply app/models/topic_hot_score.rb:10]: latest iteration handles this :hugs:

File: config/locales/client.en.yml (hunk 2 above)
Comment (pmusaraj) [reply config/locales/client.en.yml:4066]: Sounds good, no worries!

**Pull request merged**